src/Orleans.Transactions/DistributedTM/TransactionClock.cs

```diff
@@ -0,0 +1,29 @@
+﻿
+using System;
+
+namespace Orleans.Transactions
+{
+    public class TransactionClock
```

Comment:
I think "CausalClock" would be a better name since it describes what this thing provides, not just who uses it.